File: Data Structures & Algorithms/non-cyclical-number/submission-2.py
class Solution:
    def isHappy(self, n: int) -> bool:
        cache = set()
        res = n

        def calculate(r: int):
            a = str(r)
            r = 0
            for c in a:
                r += int(c) ** 2
            return r

        while res != 1:
            if res not in cache:
                cache.add(res)
                res = calculate(res)
            else:
                return False

        return True

File: Data Structures & Algorithms/anagram-groups/submission-2.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap<>();

        for (String s : strs)
        {
            char [] array = s.toCharArray();
            Arrays.sort(array);

            if (!groups.containsKey(new String(array))) {
                List<String> list = new ArrayList<>();
                list.add(s);
                groups.put(new String(array), list);
            } else {
                groups.get(new String(array)).add(s);
            }
        }

        List<List<String>> res = new ArrayList<>();
        for (String s : groups.keySet())
        {
            res.add(groups.get(s));
        }

        return res;
    }
}
